### PR TITLE
Dpp 421 test payment type breakdown

### PR DESCRIPTION
**Pull request by msbaig**

### CODE REVIEW

File: features/provider_due_amount.feature [hunk 1]
```diff
@@ -13,3 +13,17 @@ Examples:
...
 
+
+Scenario Outline: Payment type breakdown
+Given the account has a balance of <balance>
+When payment of <due amount> is due
+Then a levy payment of <levy amount> is made
+And a government payment of <government amount> is made
+And a employer payment of <employer amount> is expected
+Examples:
+| balance | due amount | levy amount | government amount | employer amount |
```
Comment (benilovj) [inline features/provider_due_amount.feature]: the wording on these labels isn't quite right, I think.

Comment (msbaig) [inline features/provider_due_amount.feature]: Wordings are as per the story DPP-421 so if anything needs changing, can you please update the story and I'll update the specs.

Comment (benilovj) [inline features/provider_due_amount.feature]: Here's the suggested language:

```
Scenario Outline: Payment type breakdown
Given an employer levy balance of <employer levy balance>
When a payment of <due amount> is due
Then the employer levy account is debited by <levy account debit>
And the provider is paid <paid by SFA> by the SFA
And the provider is due <payment due from employer> from the employer

Examples:
| employer levy balance | due amount | levy account debit | paid by SFA | payment due from employer |
| 99999                 | 1000       | 1000               | 0           | 0                         |
| 500                   | 1000       | 500                | 450         | 50                        |
| 0                     | 1000       | 0                  | 900         | 100                       |
```

Comment (sbulluk) [inline features/provider_due_amount.feature]: Broadly seems good to me. Couple of points:

1. Is it misleading to say the employer account is debited at this point? This is not strictly true
2. We should stop referring to a levy account. The current feeling is that it is an employer account. We should probably stick with levy account for now, so keep it consistent; but plan a wholesale re-wording once content clan have fully settled on the name

Comment (benilovj) [inline features/provider_due_amount.feature]: > Is it misleading to say the employer account is debited at this point? This is not strictly true

There isn't really a temporal element to this test – is this inaccuracy an issue?

> but plan a wholesale re-wording once content clan have fully settled on the name

agreed. I'd prefer to keep the naming consistent at least for the moment.

File: src/SFA.DAS.Payments.AcceptanceTests/StepDefinitions/EarningAndPaymentsSteps.cs [hunk 2]
```diff
@@ -574,20 +501,221 @@ public void ThenAPaymentIsDue(decimal dueAmount)
...
+                Assert.AreEqual(dueAmount, earning.AmountDue, $"Expected earning of {dueAmount} for period R01 but found {earning.AmountDue}");
             }
             else
             {
                 Assert.IsNull(earning, $"There was no expected earning for the period but earnigs data found");
 
             }
         }
+
+        #region Payment Type Breakdown
```
Comment (sbulluk) [inline src/SFA.DAS.Payments.AcceptanceTests/StepDefinitions/EarningAndPaymentsSteps.cs]: The steps in this region should probably be put in a new step definitions class.

**Pull request merged**